src/Mapper.cpp

```diff
@@ -302,3 +302,44 @@ void Mapper::countGates(decltype(qcMapped.cbegin())      it,
     }
   }
 }
+std::size_t Mapper::insertGate(const Edge& edge, qc::StandardOperation* op) {
+  qcMapped.emplace_back<qc::StandardOperation>(
+      qcMapped.getNqubits(), qc::Control{static_cast<qc::Qubit>(edge.first)},
+      edge.second, op->getType(), op->getParameter().at(0),
+      op->getParameter().at(1), op->getParameter().at(2));
+  return 1;
+}
+
+std::size_t Mapper::insertFlippedGate(const Edge& edge, qc::StandardOperation* op) {
+  int swapType = -1;
+  switch (op->getType()) {
+  case qc::X:
+    swapType = 1;
+    break;
+  case qc::Z:
+    swapType = 0;
+    break;
+  default:
+    swapType = -1;
+  }
```

Comment:
I did it like this because the same case distinction would be required in the cost functions as well. If you have the same switch case in multiple functions, you can easily introduce bugs if you don't keep them the same all the time. That's why I would suggest to move this case distinction to its own function. Could also add a new enum , e.g. 
```c++
enum GateFlipStrategy {
    Swap,
    Identity,
    Hadamard
}
```
to make it more clear what the return value is supposed to mean.

Reply:
Agreed! I am always in favor of splitting things up into small functions 👍🏼 
An Enum would probably be a good idea.